omniledger/service/contracts.go

```diff
@@ -163,35 +168,80 @@ func (s *Service) ContractConfig(cdb CollectionView, tx Instruction, coins []Coi
 				DarcID:     tx.ObjectID.DarcID,
 				InstanceID: OneNonce,
 			}, ContractConfigID, configBuf),
-	}, nil, nil
+	}, c, nil
 }
 
 // ContractDarc accepts the following instructions:
 //   - Spawn - creates a new darc
 //   - Invoke.Evolve - evolves an existing darc
 func (s *Service) ContractDarc(coll CollectionView, tx Instruction,
 	coins []Coin) ([]StateChange, []Coin, error) {
-	if tx.getType() != invokeType {
-		return nil, nil, errors.New("Darc can only be invoked (evolved)")
-	}
-	if tx.Invoke.Command == "evolve" {
-		darcBuf := tx.Invoke.Args.Search("darc")
-		newD, err := darc.NewDarcFromProto(darcBuf)
-		if err != nil {
-			return nil, nil, err
+	switch {
+	case tx.Spawn != nil:
+		if tx.Spawn.ContractID == ContractDarcID {
+			darcBuf := tx.Spawn.Args.Search("darc")
+			d, err := darc.NewDarcFromProto(darcBuf)
+			if err != nil {
+				return nil, nil, errors.New("given darc was not valid: " + err.Error())
```

Comment:
maybe more clear: "could not be decoded"

in the context of darc, Valid() is about the signatures, right?